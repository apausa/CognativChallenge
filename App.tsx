import React, { Suspense } from 'react';
import { RecoilRoot } from 'recoil';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';

import Home from "./src/screens/home/Home";
import Loading from "./src/screens/loading/Loading";
import Details from "./src/screens/details/Details";

export default function App() {
  const { Screen, Navigator } = createNativeStackNavigator();
  return (
    <RecoilRoot>
      <Suspense fallback={<Loading />}>
        <NavigationContainer>
          <Navigator initialRouteName="Home">
            <Screen name="Home" component={Home}/>
            <Screen name="Details" component={Details}/>
          </Navigator>
        </ NavigationContainer>
      </Suspense>
    </RecoilRoot>
  );
}
