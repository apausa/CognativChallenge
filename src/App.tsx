import React, { Suspense } from 'react';
import { RecoilRoot } from 'recoil';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';

import Home from "./screens/home/Home";
import Loading from "./screens/loading/Loading";
import Details from "./screens/details/Details";

export default function App() {
  const { Screen, Navigator } = createNativeStackNavigator();
  return (
    <RecoilRoot>
      <Suspense fallback={<Loading />}>
        <NavigationContainer>
          <Navigator initialRouteName="Home">
            <Screen name="Home" component={Home}/>
            <Screen name="Details" component={Details}/>
          </Navigator>
        </ NavigationContainer>
      </Suspense>
    </RecoilRoot>
  );
}
