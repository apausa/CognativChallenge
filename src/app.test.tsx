import React from 'react';
import { render } from '@testing-library/react-native';

import App from './App';

describe('Given an App component', () => {
  describe('When is invoked', () => {
    const screen = render(<App />);
    test('Then matches the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
  });
});