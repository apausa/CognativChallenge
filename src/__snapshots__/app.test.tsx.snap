// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an App component When is invoked Then matches the snapshot 1`] = `
<Text>
  Loading
</Text>
`;
